ProjFS.Mac/PrjFS.xcodeproj/project.pbxproj

```diff
@@ -641,6 +652,84 @@
 /* End PBXResourcesBuildPhase section */
 
 /* Begin PBXShellScriptBuildPhase section */
+		26786AED228CD0F800F53311 /* ShellScript */ = {
```

Comment:
@pmj @nickgra I didn't like updating each target to have call `GeneratePrjFSVersionHeader.sh` if the header is missing, but I couldn't find a better way of making sure that building any of the targets in Xcode would succeed if the version header was missing.

Any suggestions?

Reply:
You could create an otherwise empty target (File -> New -> Target… -> "Aggregate") and put your shell script in there. Then mark this new target as a dependency for all other relevant targets.

Reply:
Additionally, I'd list the generated header file under output paths. I'm not entirely clear on where the version number is coming from, but if it's at all related to the contents of some file(s), then such files should be listed under input files so the shell script is only run when those files change.

Reply:
> I'm not entirely clear on where the version number is coming from, but if it's at all related to the contents of some file(s), then such files should be listed under input files so the shell script is only run when those files change.

The version number comes from the Azure Pipelines build.  When it calls the build script it passes in the version:

```
Scripts/Mac/BuildGVFSForMac.sh $(BuildConfiguration) $(MacMajorAndMinorVersion).$(GVFSRevision)
```

And when a developer is building VFS4G on their machine by default we want to use "0.2.173.2" as the version.

I'm glad you brought this up because I noticed the "Mac XCode build kext" step of the release build is *not* using the build script.  It's calling XCode directly, and so that will need to be updated. 

@jrbriggs @nickgra do you know why that release build definition is not using `ProjFS.Mac/Scripts/Build.sh` to build the kext?

Reply:
> Additionally, I'd list the generated header file under output paths. I'm not entirely clear on where the version number is coming from, but if it's at all related to the contents of some file(s), then such files should be listed under input files so the shell script is only run when those files change.

I added `${BUILD_ROOT}/../../PrjFSVersion.h` as an input path to the "Run Script" step (in a new aggregate target).

Listing the same path as both input and output caused some circular dependency issues.

The script in XCode only needs to be run if `${BUILD_ROOT}/../../PrjFSVersion.h` is missing, and so picking "input" over "output" seemed like the better choice, but let me know if you disagree.